src/crimson/os/seastore/transaction_manager.h

```diff
@@ -179,7 +179,14 @@ class TransactionManager : public ExtentCallbackInterface {
     auto v = pin->get_logical_extent(t);
     if (v.has_child()) {
       return v.get_child_fut().safe_then([](auto extent) {
-	return extent->template cast<T>();
+        LOG_PREFIX(TransactionManager::read_pin);
+        SUBTRACET(seastore_tm, "found child: {}", t, *extent);
+        if (extent->is_fully_loaded()) {
```

Comment:
Not fully loaded extents have been loaded in pin->get_logical_extent(t) before this, see https://github.com/aisakaki/ceph/blob/f19de7c77f7d6b783802f6c18e2e9fe106c90243/src/crimson/os/seastore/cache.h#L555

Reply:
get_logical_extent returns an union  (child_pos_t，future<extent>).
when come to v.has_child(), the future might not be resolved, so load not-fully-loaded extent here after get_logical_extent might break the atomicity